src/rulesets/LinearAlgebra/structured.jl

```diff
@@ -90,64 +90,55 @@ end
 ##### `Adjoint`
 #####
 
-# ✖️✖️✖️TODO: Deal with complex-valued arrays as well
-function rrule(::Type{<:Adjoint}, A::AbstractMatrix{<:Real})
-    function Adjoint_pullback(ȳ)
-        return (NO_FIELDS, adjoint(ȳ))
-    end
+function rrule(::Type{<:Adjoint}, A::AbstractMatrix{<:Union{Real, Complex}})
```

Comment:
The `Adjoint`/`Transpose` functions in base work even with non-real/complex types, e.g. `Quaternion`s. Given that the pushforwards and pullbacks just call adjoint themselves and there, aren't these rules a good example of where it is safe to drop the type constraint?

Reply:
Turns out I had forgotten to account for `rrule(adjoint, ::Adjoint{T, Vector{T})` (and for `transpose`). Tests are now in, so hopefully this is now fixed 🤞

Reply:
I don't follow. There is still from what I can tell not a rule for  `rrule(::Type{<:Adjoint}, A::AbstractMatrix{T}) where {T<:Number}`

Reply:
🤦 I mean to reply to your comment about Zygote integration errors. I should probably go to bed...

You're correct, I've still not resolved this.

Reply:
I've pushed a change. Should be sorted.